src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -668,3 +673,22 @@
         "  where pg_stat_activity.datname = ?"
         "    and pg_stat_activity.usename = ?")
    [db user]))
+
+(defn revoke-role-db-access
+  [role db]
+  ;; revoke commands can't be parameterized with the pgjdbc driver right now
+  (prn :revoke (format "revoke connect on database %s from %s restrict"
+                       (double-quote db) (double-quote role)))
+  (do-commands
+   (format "revoke connect on database %s from %s restrict"
```

Comment:
I'm wrong about this... this func is called using the migrations pool which is set up using the migration user. Leaving the comments here for a reminder in case I forget :)